bin/createInhouseSamplesheetFromGS.py

```diff
@@ -0,0 +1,242 @@
+                                new_row.append(row[header])
+
+                        #Add newly build row to to total csv list.
+                        new_rows_list.append(new_row)
+
+                f1.close()
+
+        # Do the writing of new samplesheet tot outputPath.
+        newfilename = samplesheetOutputDir + basename(projectSamplesheetPath)
+        logger.write("outputFile:" + newfilename + "\n")
+        with open(newfilename, 'w') as f2:
+                writer = csv.writer(f2, delimiter = ',')
+                writer.writerows(new_rows_list)
+        f2.close()
+
+
+def makeOriginalFilenameHashmap(md5file):
+
+
+        flowcellDict = defaultdict(dict)
+
+        #get sequencerundir information.
+        for root, dirs, files in os.walk(args.GenomeScanInputDir):
+                for dir in dirs:
+
+                        if re.match("([0-9]{6})_([a-zA-Z0-9]{6})_([0-9]{4})_([a-zA-Z0-9]{9})$", dir):
+                                m = re.match("^([0-9]{6})_([a-zA-Z0-9]{6})_([0-9]{4})_([a-zA-Z0-9]{9})$", dir)
+
+                                runStartDate = m.group(1)
+                                sequencer = m.group(2)
+                                runId = m.group(3)
+                                flowcellId = m.group(4)
+
+                                flowcellDict[flowcellId]=(runStartDate,sequencer,runId,flowcellId)
+
+                        else:
+                                logger.write(dir + " is not the converted sequence outputdir.\n")
+
+        originalFileNameDict = defaultdict(dict)
+
+        #parse original fileNames, and combine with sequence run dir.
+        for line in md5file:
+
+                if re.match("^([a-z0-9]+).+?([a-zA-Z0-9]{9})_([0-9-]+)_([ATGCN]+-[ATGCN]+)_L00([0-9]{1}).+R1.fastq.gz$",line):
+                        m = re.match("^([a-z0-9]+).+?([a-zA-Z0-9]{9})_([0-9-]+)_([ATGCN]+-[ATGCN]+)_L00([0-9]{1}).+R1.fastq.gz$", line)
+                        flowcellId = m.group(2)
+                        GS_ID = m.group(3)
+                        barcodes = m.group(4)
+                        lane = m.group(5)
+
+
+                        originalFileNameDict.setdefault(barcodes + '-' + GS_ID, []).append(lane)
+                        originalFileNameDict.setdefault(barcodes + '-' + GS_ID, []).append(barcodes)
+                        originalFileNameDict.setdefault(barcodes + '-' + GS_ID, []).append(flowcellId)
+                        originalFileNameDict.setdefault(barcodes + '-' + GS_ID, []).append(flowcellDict[flowcellId][0])
+                        originalFileNameDict.setdefault(barcodes + '-' + GS_ID, []).append(flowcellDict[flowcellId][1])
+                        originalFileNameDict.setdefault(barcodes + '-' + GS_ID, []).append(flowcellDict[flowcellId][2])
+
+        md5file.close()
+        return originalFileNameDict
+
+
+#
+## Main
+#
+
+#get commandline parameters
+parser = argparse.ArgumentParser(description='Process some integers.')
```

Comment:
"Process some integers"?
I don't see a -h/--help option o does Python provide this automagically?

Reply:
Yes, that works out of the box.